From: Levi Yun <ppbuk5246@gmail.com>
Subject: damon/sysfs: Fix possible memleak on damon_sysfs_add_target.
Date: Sun, 25 Sep 2022 23:02:57 +0900

When damon_sysfs_add_target couldn't find proper task, New allocated
damon_target structure isn't registered yet, So, it's impossible to free
new allocated one by damon_sysfs_destroy_targets.

By calling additional damon_free_target when find_get_pid function, Fix
possible memory leak.

Link: https://lkml.kernel.org/r/20220925140257.23431-1-ppbuk5246@gmail.com
Signed-off-by: Levi Yun <ppbuk5246@gmail.com>
Cc: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/damon/sysfs.c~damon-sysfs-fix-possible-memleak-on-damon_sysfs_add_target
+++ a/mm/damon/sysfs.c
@@ -2174,8 +2174,11 @@ static int damon_sysfs_add_target(struct
 		return -ENOMEM;
 	if (damon_target_has_pid(ctx)) {
 		t->pid = find_get_pid(sys_target->pid);
-		if (!t->pid)
+		if (!t->pid) {
+			damon_free_target(t);
+
 			goto destroy_targets_out;
+		}
 	}
 	damon_add_target(ctx, t);
 	err = damon_sysfs_set_regions(t, sys_target->regions);
_
